<launch>

<arg name="first_camera_name" default="cam0" />
<arg name="second_camera_name" default="cam1" />
<arg name="scale" default="1.0" />
<arg name="process_every_nth_frame" default="1" />
<arg name="stereo_params_camchain" default="/home/z/catkin_ws/src/mav_tools/mav_startup/parameters/mavs/ibis/camchain_p23023.yaml"/>



  <node name="stereo_undistort" pkg="image_undistort" type="stereo_undistort_node">
    <param name="input_camera_info_from_ros_params" value = "true"/>
    <param name="first_camera_namespace" value="$(arg first_camera_name)"/>
    <param name="second_camera_namespace" value="$(arg second_camera_name)"/>
    <param name="scale" value="$(arg scale)"/>
    <param name="process_every_nth_frame" value="$(arg process_every_nth_frame)"/>
    
    <rosparam file="$(arg stereo_params_camchain)"/>

    <remap from="raw/first/image" to="$(arg first_camera_name)/image_raw"/>
    <remap from="raw/second/image" to="$(arg second_camera_name)/image_raw"/>
  </node>


</launch>